### PR TITLE
solution

### PR DESCRIPTION
**Pull request by liliya-dev**

### CODE REVIEW

File: src/arrayMethodIndexOf.js [hunk 1]
```diff
@@ -6,8 +6,30 @@
...
+        : reassignedIndex = this.length + fromIndex;
+    } else if (fromIndex > this.length - 1) {
+      return -1;
+    }
+
+    for (let i = reassignedIndex; i < this.length; i++) {
+      if (!Number.isNaN(searchElement)) {
+        if (this[i] === searchElement) {
+          return i;
+        }
```
Comment (nazmeln) [inline src/arrayMethodIndexOf.js:22]: u can use Object.is()

File: src/arrayMethodIndexOf.js [hunk 2]
```diff
@@ -6,8 +6,30 @@
...
+      } else {
+        if (Number.isNaN(this[i])) {
+          return i;
+        }
```
Comment (nazmeln) [inline src/arrayMethodIndexOf.js:26]: same

File: src/arrayMethodIndexOf.js [hunk 3]
```diff
@@ -6,8 +6,30 @@
...
-    // write code here
+  [].__proto__.indexOf2 = function(searchElement, fromIndex = 0) {
+    let reassignedIndex = fromIndex;
+
+    if (fromIndex < 0) {
+      (Math.abs(fromIndex) > this.length)
+        ? reassignedIndex = 0
+        : reassignedIndex = this.length + fromIndex;
+    } else if (fromIndex > this.length - 1) {
+      return -1;
```
Comment (nazmeln) [inline src/arrayMethodIndexOf.js]: u can return -1 in the end

File: src/arrayMethodIndexOf.js [hunk 4]
```diff
@@ -6,8 +6,30 @@
...
-  [].__proto__.indexOf2 = function(searchElement, fromIndex) {
-    // write code here
+  [].__proto__.indexOf2 = function(searchElement, fromIndex = 0) {
+    let reassignedIndex = fromIndex;
+
+    if (fromIndex < 0) {
+      (Math.abs(fromIndex) > this.length)
+        ? reassignedIndex = 0
+        : reassignedIndex = this.length + fromIndex;
+    } else if (fromIndex > this.length - 1) {
```
Comment (nazmeln) [inline src/arrayMethodIndexOf.js]:
```suggestion
 if (reassignedIndex < 0) {
      reassignedIndex += this.length;
    }

    if (reassignedIndex < 0) {
      reassignedIndex = 0;
    }
```

**Review by nazmeln — CHANGES_REQUESTED** [review]

File: src/arrayMethodIndexOf.js (hunk 1 above)
Comment (liliya-dev) [reply src/arrayMethodIndexOf.js:22]: why i can't use !Number.isNaN()?

Comment (nazmeln) [reply src/arrayMethodIndexOf.js:22]: Okay :)

**Review by nazmeln — APPROVED** [review]

**Review by mriynyk — APPROVED** [review]

**Pull request closed without merging**